src/midend/programTransformation/ompLowering/omp_simd.cpp

```diff
@@ -700,41 +740,42 @@ void OmpSupport::transOmpSimd(SgNode *node) {
     SageInterface::forLoopNormalization(for_loop);
     
     // Create the SIMD compiler object
-    OmpSimdCompiler *cc = new OmpSimdCompiler;
-    cc->for_loop = for_loop;
-
-    // Now work on the body. Run the first two passes
-    SgBasicBlock *new_block = SageBuilder::buildBasicBlock();
-    Rose_STL_Container<SgNode *> *ir_block = new Rose_STL_Container<SgNode *>();
-    
     bool isArm = false;
     if (simd_arch == ArmAddr3 || simd_arch == Arm_SVE2) isArm = true;
     
-    if (!omp_simd_pass1(cc, target, for_loop, new_block, isArm)) {
-        delete ir_block;
+    OmpSimdCompiler *cc = new OmpSimdCompiler(isArm);
+    cc->setTarget(target);
+    cc->setForLoop(for_loop);
+
+    // Now work on the body. Run the first two passes
+    //SgBasicBlock *new_block = SageBuilder::buildBasicBlock();
+    //Rose_STL_Container<SgNode *> *ir_block = new Rose_STL_Container<SgNode *>();
```

Comment:
Similarly, if this piece of code won't be used and they are not kept for some records, we could remove them as well.